internal/db/delegation.go

```diff
@@ -180,11 +184,13 @@ func buildAdditionalDelegationFilter(
 	baseFilter primitive.M,
 	filters *DelegationFilter,
 ) primitive.M {
-	if filters.States != nil {
-		baseFilter["state"] = bson.M{"$in": filters.States}
-	}
-	if filters.AfterTimestamp != 0 {
-		baseFilter["staking_tx.start_timestamp"] = bson.M{"$gte": filters.AfterTimestamp}
+	if filters != nil {
+		if filters.States != nil {
+			baseFilter["state"] = bson.M{"$in": filters.States}
+		}
+		if filters.AfterTimestamp != 0 {
+			baseFilter["staking_tx.start_timestamp"] = bson.M{"$gte": filters.AfterTimestamp}
+		}
 	}
```

Comment:
I suppose we can avoid embedded ifs by returning the `baseFilter` first if `filters` is nil